### PR TITLE
V8 port account_move_line_search_extension

### PR DESCRIPTION
**Pull request by luc-demeyer**
@StefanRijnhart 
stephan, there is some overlap between the functionality offered by the 'account_move_line_search_extension' module and the Therp 'account_move_line_no_default_search'.
Should we keep both ?

### CODE REVIEW

**Comment by antespi** [comment]
:+1:

File: account_move_line_search_extension/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+##############################################################################
+
+{
+    'name': 'Journal Items Search Extension',
+    'version': '0.2',
+    'license': 'AGPL-3',
+    'author': 'Noviat',
+    'category': 'Generic Modules',
+    'description': """
+Journal Items Search Extension
```
Comment (pedrobaeza) [inline account_move_line_search_extension/__openerp__.py]: Please extract this description to a file README.rst.

**Comment by pedrobaeza** [comment]
The new path for icons in v8 is /static/description/icon.png, so please move image there.

**Comment by pedrobaeza** [comment]
Testing the module, I see a side effect/issue that prevents to change normally from form view to list, and views icons dissapear. Can you please check it?

Another great improvement/differentiation of this module from the one by Therp is to add a balance column that allows to see directly this data.

**Comment by luc-demeyer** [comment]
The latest update contains a number of enhancements and bugfixes:
- support for search on analytic account
- code restructured to allow adding fields via inherited module
- partner search fixed to support also inactive partners
- removal of code that became obsolete after turning the view into read-only

File: account_move_line_search_extension/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Journal Items Search Extension',
+    'version': '0.6',
+    'license': 'AGPL-3',
+    'author': 'Noviat',
```
Comment (adrienpeiffer) [inline account_move_line_search_extension/__openerp__.py]: Can you add OCA as author ?

**Comment by adrienpeiffer** [comment]
A little on the manifest. Otherwise :+1: (Code review and functionnal test)

**Comment by chrisdec** [comment]
:+1:  (functional tests)

**Comment by pedrobaeza** [comment]
I have detected that account search only matches the start of the account, not intermediate values. Is that intended?

What about my suggestion of adding the balance column? Tax amount and code will be also very welcome

**Comment by luc-demeyer** [comment]
@pedrobaeza 

1) account search on start: this is intended : 
accountants tend to know by hard the first few characters of a general account when performing searches and they can complete the drill-down by entering more digits when too many lines (accounts) are displayed.

2) adding balance column is  easy, but maybe better in a separate module. I can make the "https://apps.openerp.com/apps/modules/8.0/account_move_line_balance" module dependent on "account_move_line_search extension", add the "absolute balance" field after the last search field and submit this module to the OCA.

3) concerning tax amount & code : I haven't seen this as a general requirement in our customer base (probably since we always install the 'account_journal_report_xls' module which allows easy XLS controls on taxes). 

Concerning extra fields: it's easy to add more fields via a custom module on a case by case basis: just a few lines of javascript code and a t-extend on the qweb template.

I personally think this PR is ready for merge in it's current form (technical and functional reviews are ok).

We can add more fields/functionality via new modules or new PR's.

**Comment by pedrobaeza** [comment]
OK, I agree to merge this one, but please move _account_move_line_balance_ to OCA ASAP.

:+1:

**Comment by sbidoul** [comment]
@pedrobaeza @luc-demeyer regarding the balance field there is already #183 which adds such field.

**Comment by pedrobaeza** [comment]
Cool! I didn't remember it

**Comment by luc-demeyer** [comment]
@sbidoul 
For the use cases in my customer base, the account_move_line_balance module is a far better "reconcile helper" since it displays the absolute balance allowing to get equal (or nearly equal) debit/credit fields close to each other via a single click. On the other hand PR #183 shows totals (similar to what we used to implement in our V6 and V7 customer projects). 
Idieally, we merge both approaches together (allowing sort by absolute value but keeping the sum of the debit/credit). On top of such a combined module we make it dependent on the account_move_line_search extension to add the balance to the search fields.
As an alternative we can consider Pedro's suggestion to bundle it all in one module (something like 'account_move_line_search_and_reconcile_helper').

**Comment by sbidoul** [comment]
Hi @luc-demeyer,

Yes I remember we discussed the absolute balance use case during the sprint.

I think the fields that needs to be added and displayed in the tree views is the balance as debit - credit (not absolute) as it allows to see the total for complex multiple reconciliation.

Your use case for absolute balance should done with a custom sort in account_move_reconcile_helper (`order by abs(debit-credit)` or something). But I'm not sure yet how to implement it in Odoo.

I'm fine with account_move_line_search_extension depending on reconcile helper, or a combined module on top of both.

cc/ @adrienpeiffer @pedrobaeza

**Pull request merged**